includes/razorpay-subscriptions.php

```diff
@@ -0,0 +1,372 @@
+
+        //
+        // If new plan was created, we delete the old plan id
+        // If we created a new planId, we have to store it as post metadata
+        //
+        if ($created === true)
+        {
+            delete_post_meta($productId, $key);
+
+            add_post_meta($productId, $key, $planId, true);
+        }
+
+        return $planId;
+    }
+
+    /**
+     * Takes in product metadata and product
+     * Creates or gets created plan
+     *
+     * @param $metadata,
+     * @param $product
+     *
+     * @return string $planId
+     * @return bool $created
+     */
+    protected function createOrGetPlanId($metadata, $product)
+    {
+        $planArgs = $this->getPlanArguments($product);
+
+        $currency = get_woocommerce_currency();
+
+        $key = self::RAZORPAY_PLAN_ID . '_'. strtolower($currency);
+
+        //
+        // If razorpay_plan_id is set in the metadata,
+        // we check if the amounts match and return the plan id
+        //
+        if (isset($metadata[$key]) === true)
+        {
+            $planId = $metadata[$key][0];
+
+            try
+            {
+                $plan = $this->api->plan->fetch($planId);
+            }
+            catch (Exception $e)
+            {
+                //
+                // If plan id fetch causes an error, we re-create the plan
+                //
+                $planId = $this->createPlan($planArgs);
+
+                return array($planId, true);
+            }
+
+            if ($plan['item']['amount'] === $planArgs['item']['amount'])
+            {
+                return array($plan['id'], false);
+            }
+        }
+
+        //
+        // By default we create a new plan
+        // if metadata doesn't have plan id set
+        //
+        $planId = $this->createPlan($planArgs);
+
+        return array($planId, true);
+    }
+
+    protected function createPlan($planArgs)
+    {
+        try
+        {
+            $plan = $this->api->plan->create($planArgs);
+        }
+        catch (Exception $e)
+        {
+            $message = $e->getMessage();
+
+            throw new Errors\Error(
+                $message,
+                WooErrors\ErrorCode::API_PLAN_CREATION_FAILED,
+                400
+            );
+        }
+
+        // Storing the plan id as product metadata, unique set to true
+        return $plan['id'];
+    }
+
+    protected function getPlanArguments($product)
+    {
+        $productId = $product['product_id'];
+
+        $period       = WC_Subscriptions_Product::get_period($productId);
+        $interval     = WC_Subscriptions_Product::get_interval($productId);
+        $salePrice    = get_post_meta($productId)['_sale_price'][0];
+        $recurringFee = WC_Subscriptions_Product::get_price($productId);
+
+        //
+        // If the item is on sale, sell it for the sale price
+        //
+        if (empty($salePrice) === false)
+        {
+            $recurringFee = $salePrice;
```

Comment:
@mayankamencherla `get_price` takes care of this for you: https://github.com/wp-premium/woocommerce-subscriptions/blob/master/includes/class-wc-subscriptions-product.php#L384-L404